src/pages/home/report/ReportActionsView.tsx

```diff
@@ -87,10 +119,75 @@ function ReportActionsView({
             return;
         }
 
-        Report.openReport(reportID);
+        Report.openReport(reportID, reportActionID);
     };
 
+    useLayoutEffect(() => {
+        setCurrentReportActionID('');
+    }, [route]);
+
+    const listID = useMemo(() => {
+        isFirstLinkedActionRender.current = true;
+        const newID = generateNewRandomInt(listOldID, 1, Number.MAX_SAFE_INTEGER);
+        listOldID = newID;
+        return newID;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [route, isLoadingInitialReportActions]);
+
+    const indexOfLinkedAction = useMemo(() => {
+        if (!reportActionID || isLoading) {
+            return -1;
+        }
+
+        return allReportActions.findIndex((obj) => String(obj.reportActionID) === String(isFirstLinkedActionRender.current ? reportActionID : currentReportActionID));
+    }, [allReportActions, currentReportActionID, reportActionID, isLoading]);
+
+    const reportActions = useMemo(() => {
+        if (!reportActionID) {
+            return allReportActions;
+        }
+        if (isLoading || indexOfLinkedAction === -1) {
+            return [];
+        }
+
+        if (isFirstLinkedActionRender.current) {
+            return allReportActions.slice(indexOfLinkedAction);
+        }
+        const paginationSize = getInitialPaginationSize;
+        return allReportActions.slice(Math.max(indexOfLinkedAction - paginationSize, 0));
+        // currentReportActionID is needed to trigger batching once the report action has been positioned
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [reportActionID, allReportActions, indexOfLinkedAction, isLoading, currentReportActionID]);
+
+    const hasMoreCached = reportActions.length < allReportActions.length;
+    const newestReportAction = useMemo(() => reportActions?.[0], [reportActions]);
+    const handleReportActionPagination = useCallback(
+        ({firstReportActionID}: {firstReportActionID: string}) => {
+            // This function is a placeholder as the actual pagination is handled by visibleReportActions
+            if (!hasMoreCached) {
+                isFirstLinkedActionRender.current = false;
+                fetchNewerAction(newestReportAction);
+            }
+            if (isFirstLinkedActionRender.current) {
+                isFirstLinkedActionRender.current = false;
+            }
+            setCurrentReportActionID(firstReportActionID);
+        },
+        [fetchNewerAction, hasMoreCached, newestReportAction],
+    );
+
+    const mostRecentIOUReportActionID = useMemo(() => ReportActionsUtils.getMostRecentIOURequestActionID(reportActions), [reportActions]);
+    const hasCachedActionOnFirstRender = useInitialValue(() => reportActions.length > 0);
+    const hasNewestReportAction = reportActions[0]?.created === report.lastVisibleActionCreated;
```

Comment:
I think we should consider the visible reportActions here to avoid issue https://github.com/Expensify/App/issues/52861